src/api/app/controllers/webui/projects/rebuild_times_controller.rb

```diff
@@ -18,7 +18,12 @@ def show
           return
         end
         bdep = Backend::Api::BuildResults::Binaries.builddepinfo(@project.name, @repository, @arch)
-        jobs = Backend::Api::BuildResults::JobHistory.not_failed(@project.name, @repository, @arch, (@packages.size + @ipackages.size) * 3)
+        jobs = Backend::Api::BuildResults::JobHistory.for_repository_and_arch(project_name: @project.name,
+                                                                              repository_name: @repository,
+                                                                              arch_name: @arch,
+                                                                              filter: { limit: (@packages.size + @ipackages.size) * 3,
```

Comment:
I would move the filter block to a private method and have the 3 as default parameter in the method, something like

```
filter: set_limit

... 

private
def set_limit(limit_offset=3)
  {  limit: (@packages.size + @ipackages.size) * limit_offset,
     code: ['succeeded', 'unchanged'] 
  }
end
```

Reply:
for readability, I'm old I cannot cope with long lines